### ISSUE TITLE
bug:  svgr-rs does not support unicode chars

### ISSUE THREAD

**Issue by stormslowly** [umijs/mako#1600]
## problem 
```svg
...
<title>&amp;中文</title>
...
```

```txt
panicked at svgr-rs/src/hast_to_swc_ast/decode_xml.rs:60:21:
byte index 6 is not a char boundary; it is inside '文' (bytes 5..8) of `x&中文`
```

## solution 
we don't have to escape by [hand](https://github.com/svg-rust/svgr-rs/blob/main/crates/core/src/hast_to_swc_ast/decode_xml.rs#L1), use the crate [html-escape](https://crates.io/crates/html-escape) 

will be fixed in pr #1444

**Comment by stormslowly** [comment]
the root cause is below
svgr-rs 
1. use swc_xml_parser the svg content , in the html ast. title tag's text children content become `&中文` [ref](https://github.com/swc-project/swc/blob/85cc2bd79c3193cb0a8b54e4fce0efc1aa15b271/crates/swc_xml_parser/src/lexer/mod.rs#L395-L436)
2. then svgr-rs trying to convert the html ast to swc jsx ast,  html entity is concerned, but doing by parsing it by bytes `as_bytes()`
3.  svgr-rs peeks 2 or 4 bytes after `&` to convert them to string , then hit `byte index 6 is not a char boundary` the error

IMO, the best way to solve the problem is , swc_xml_parser un-escaping the text children of html

**Comment by stormslowly** [comment]
besides this problem， Mako should put  `compile()` in a [catch_unwind](https://doc.rust-lang.org/std/panic/fn.catch_unwind.html)
in case this problem,  node process should get an error other than  process abort.

**Comment by SyMind** [comment]
@stormslowly I have published version 0.2.0 to incorporate your PR.
